### DISCUSSION TITLE
Route diagram display waypoint names

### DISCUSSION THREAD

**Discussion by DIrgi** [devemux86/cruiser#505] [Q&A] [answered]
I recently posted that request/idea in another thread. Emux suggested to extract it into a stand alone.
So, I just copy & paste my request and his first answer:
<hr>

![image](https://github.com/devemux86/cruiser/assets/57968526/05277052-42ea-4837-a4f4-4238072652b2)

Question: Will it be possible to either/or (or even permanent for all visible waypoints on demand) by tapping on a waypoint in the diagram to see the name. I did an example construction how it can look like.

I know it'll be small, especially when you have a small smartphone. On a tablet it'll work. However in the example you see the name box examples popping up by tapping on a waypoint or are even visible the whole time (on settings demand as mentioned). A good orientation for the box can be, when the waypoint sign is below the middle line, the box appears above, when the symbol is above, then the box appears below.
What do you think?

_

> Emux answer:
> It also depends on what the diagram allows to implement, as this is not a map library with rich overlays.
> So touchable markers or custom labels may be too much for it. 🙂

_

**Comment by DIrgi** [comment]
Some additional thoughts from my side. Since we have now the "tapping" for to diagram zoom in/out, maybe it'll not work at least on the whole diagram width - maybe just in the middle. 
Emux: you can probably check if just tapping on waypoint work to display the name on demand or if activation for permanent names in settings on demand can be a solution.
When it comes to the zoom: If tapping for waypoints will be active, maybe to zoom in/out can be managed by "tap **and** hold" to zoom automatically in/out after 1sec.

**Comment by devemux86** [comment, marked as answer]
@DIrgi thanks for the suggestion.

The diagram is not as flexible as the map view to add complex overlays and gestures.
Also if it is not full screen, then it does not have enough space to see readable labels.
